### PR TITLE
[Graph Browser 2] Use triples to get out arcs for observation nodes

### PR DESCRIPTION
**Pull request by chejennifer**
Observation nodes have no property labels or property values, so in order to get out arcs, fetch data from api/browser/triples

### CODE REVIEW

File: static/js/browser2/arc_section.tsx [hunk 1]
```diff
@@ -35,6 +35,7 @@ interface ArcSectionStateType {
   inLabels: string[];
   outLabels: string[];
   provDomain: { [key: string]: URL };
+  dataFetched: boolean;
```
Comment (beets) [inline static/js/browser2/arc_section.tsx:38]: i find prefixing boolean parameters with "is" or "has" helpful for readability, e.g. isDataFetched.

File: static/js/browser2/arc_section.tsx [hunk 2]
```diff
@@ -97,8 +98,13 @@ export class ArcSection extends React.Component<
...
           outLabels: labelsData["outLabels"],
           provDomain,
+          dataFetched: true,
         });
-      }
-    );
+      })
+      .catch(() => {
+        this.setState({
+          dataFetched: true,
```
Comment (beets) [inline static/js/browser2/arc_section.tsx:106]: i suppose it's whether the fetch attempt has completed, not that the data has been fetched. should an error message be displayed?

File: static/js/browser2/out_arc_section.tsx [hunk 3]
```diff
@@ -174,4 +179,43 @@ export class OutArcSection extends React.Component<
...
+      outArcs.forEach((outArc) => {
+        const predicate = outArc.predicate;
+        if (IGNORED_OUT_ARC_PROPERTIES.has(predicate)) {
+          return;
+        }
+        if (!outArcsByPredicateAndProvenance[predicate]) {
+          outArcsByPredicateAndProvenance[predicate] = {};
+        }
+        const provId = outArc.provenanceId;
+        if (!(provId in outArcsByPredicateAndProvenance[predicate])) {
```
Comment (beets) [inline static/js/browser2/out_arc_section.tsx:199]: before this line, consider pulling out "outArcsByPredicateAndProvenance[predicate]" to a local var, which you can reference more easily here and below

File: static/js/browser2/arc_section.tsx (hunk 1 above)
Comment (chejennifer) [reply static/js/browser2/arc_section.tsx]: done

File: static/js/browser2/arc_section.tsx (hunk 2 above)
Comment (chejennifer) [reply static/js/browser2/arc_section.tsx]: done

File: static/js/browser2/out_arc_section.tsx (hunk 3 above)
Comment (chejennifer) [reply static/js/browser2/out_arc_section.tsx]: done

File: static/js/browser2/arc_section.tsx (hunk 1 above)
Comment (chejennifer) [reply static/js/browser2/arc_section.tsx]: done

File: static/js/browser2/arc_section.tsx (hunk 2 above)
Comment (chejennifer) [reply static/js/browser2/arc_section.tsx]: done

File: static/js/browser2/out_arc_section.tsx (hunk 3 above)
Comment (chejennifer) [reply static/js/browser2/out_arc_section.tsx]: done

File: static/js/browser2/out_arc_section.tsx [hunk 4]
```diff
@@ -141,17 +147,19 @@ export class OutArcSection extends React.Component<
             if (!(predicate in outArcsByPredicateAndProvenance)) {
               outArcsByPredicateAndProvenance[predicate] = {};
             }
+            const outArcsOfPredicate =
```
Comment (shifucun) [inline static/js/browser2/out_arc_section.tsx:150]: Both "Predicate" and "Property" are used which is a bit in-consistent, might be using "property" throughout? And I feel the names here a bit too long, something like "outArcsByPropProv" might work as well.

**Review by shifucun — COMMENTED** [review]
Overall looks good! Just a minor comment.

File: static/js/browser2/out_arc_section.tsx (hunk 4 above)
Comment (chejennifer) [reply static/js/browser2/out_arc_section.tsx:150]: I think using either "predicate" or "property" will present the same problem. The data we get back from the triples api uses "predicate", but the data we get back from the propVals api uses "property". I believe this component mainly uses "predicate" right now, do you want me to change it to "property"?

Comment (shifucun) [reply static/js/browser2/out_arc_section.tsx:150]: I see. That's a good point. We should use "property" for our next version of API so the naming are aligned. Then no need to do the conversion here.

Comment (chejennifer) [reply static/js/browser2/out_arc_section.tsx:150]: ok, sounds good! But, I'll shorten the names

**Review by shifucun — APPROVED** [review]
Thanks for the update!

**Review by beets — APPROVED** [review]

**Comment by beets** [comment]
thanks for the updates!

**Pull request merged**